src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -990,8 +990,11 @@ private bool IsContractCheckMethod(IMethodSymbol method)
             return false;
         }
 
-        private bool IsDebugAssertMethod(IMethodSymbol method)
-            => Equals(method, DebugAssertMethod);
+        private bool IsDebugAssertMethod(IMethodSymbol method) =>
```

Comment:
Would it be worth to add some option to enable/disable this behavior? I absolutely understand the reason of this change but I think that depending on the context of the application we may want to be really sure of some "state". 
I am not sure if I am really clear but `Debug.Assert` gives some safety in `Debug` but after that in `Release` you no longer have any check which means you could end up in this invalid scenario.

Reply:
Not sure I understood your point, but `Debug.Assert` is for asserting expected behavior in `Debug` mode and to catch a failure in a development phase, it should/could not assert `Release`